### PR TITLE
Migrate to mime-types npm package

### PR DESCRIPTION
**Pull request by bryceitoc9**
## Description

Migrating from a curated list of MIME types to ones provided by the `mime-types` NPM package

## Motivation

Provides sane defaults for users without having to go through us.

## Testing

image
This includes some files which did not have MIME types before (.java, .log)

## Checklist

-   [x] I have read the **README** document
-   [x] I have read the **CONTRIBUTING** document
-   [x] My code follows the code style of this project
-   [x] I have added tests to cover my changes
-   [x] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/tasks/S3Upload/TaskOperations.ts [hunk 1]
```diff
@@ -171,11 +172,11 @@ export class TaskOperations {
...
                 const fileBuffer = fs.createReadStream(matchedFile)
                 try {
-                    let contentType: string | undefined
+                    let contentType: string | false | undefined
                     if (this.taskParameters.contentType) {
                         contentType = this.taskParameters.contentType
                     } else {
-                        contentType = knownMimeTypes.get(path.extname(matchedFile))
+                        contentType = lookup(path.extname(matchedFile))
                         if (!contentType) {
```
Comment (awschristou) [inline src/tasks/S3Upload/TaskOperations.ts:180]: Is there a test that covers when lookup returns false?

Comment (bryceitoc9) [reply src/tasks/S3Upload/TaskOperations.ts:180]: I can try and add one, the problem is that we have to use the project's file tree. The dotfiles in the home dir might work.

File: package-lock.json [hunk 2]
```diff
@@ -1,11136 +1,8 @@
 {
     "name": "aws-vsts-tools",
     "version": "1.10.0",
-    "lockfileVersion": 2,
```
Comment (hunterwerlla) [inline package-lock.json:4]: Upgrade your NPM so this doesn't happen

File: .changes/next-release/Feature-0958856d-08ef-47c0-b87a-d8e17ab1c1c4.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "Expanded MIME types for S3 uploads"
```
Comment (hunterwerlla) [inline .changes/next-release/Feature-0958856d-08ef-47c0-b87a-d8e17ab1c1c4.json:3]: Maybe "Detect more MIME types for S3 uploads"

File: src/tasks/S3Upload/TaskOperations.ts [hunk 4]
```diff
@@ -171,11 +172,11 @@ export class TaskOperations {
             if (!stats.isDirectory()) {
                 const fileBuffer = fs.createReadStream(matchedFile)
                 try {
-                    let contentType: string | undefined
+                    let contentType: string | false | undefined
```
Comment (hunterwerlla) [inline src/tasks/S3Upload/TaskOperations.ts:175]: I don't think this can return undefined

File: src/tasks/S3Upload/TaskOperations.ts (hunk 1 above)
Comment (hunterwerlla) [reply src/tasks/S3Upload/TaskOperations.ts:180]: Add something like ".thiswillneverresolveamimetype" that will never work

File: src/tasks/S3Upload/TaskOperations.ts (hunk 4 above)
Comment (bryceitoc9) [reply src/tasks/S3Upload/TaskOperations.ts:175]: Yeah, I think I was trying to cover for the `this.taskParameters.contentType`.

File: tests/taskTests/s3Upload/s3Upload-test.ts [hunk 5]
```diff
@@ -193,4 +194,32 @@ describe('S3 Upload', () => {
             expect(e.message).toContain('Invalid match expression =false')
         })
     })
+
+    test('A couple different MIME types', async () => {
```
Comment (hunterwerlla) [inline tests/taskTests/s3Upload/s3Upload-test.ts:198]: nit: drop the "A couple different"

File: tests/taskTests/s3Upload/s3Upload-test.ts [hunk 6]
```diff
@@ -215,9 +215,9 @@ describe('S3 Upload', () => {
         const taskParameters = { ...baseTaskParameters }
         taskParameters.createBucket = true
         taskParameters.bucketName = 'potato'
-        taskParameters.sourceFolder = path.join(__dirname, '..', '..', 'resources')
+        taskParameters.sourceFolder = path.join(__dirname, '..', '..', 'resources', 'mimeTypes')
         // as of 19 May 2021: tests/resources/echo.bat and tests/resources/codeDeployCode/test.txt
```
[the remark this answers is not in the document]
Comment (bryceitoc9) [reply tests/taskTests/s3Upload/s3Upload-test.ts]: I blame the fact that I did this in a light theme and forgot that the comment was there.

**Review by hunterwerlla — APPROVED** [review]
Light colorschemes 4 life

**Pull request merged**